File: support-utils/src/androidTest/java/com/hendraanggrian/support/utils/content/MimeTypesTest.java
package com.hendraanggrian.support.utils.content;

import android.support.test.runner.AndroidJUnit4;

import org.junit.Test;
import org.junit.runner.RunWith;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNull;

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
@RunWith(AndroidJUnit4.class)
public class MimeTypesTest {

    @Test
    public void guessExtension() throws Exception {
        assertEquals(MimeTypes.getMimeTypeFromURL("/sdcard/home/important.txt"), "text/plain");

        assertEquals(MimeTypes.getMimeTypeFromExtension("txt"), "text/plain");
        assertEquals(MimeTypes.getMimeTypeFromExtension("jpg"), "image/jpeg");
        assertNull(MimeTypes.getMimeTypeFromExtension(""));
        assertNull(MimeTypes.getMimeTypeFromExtension(null));
    }

    @Test
    public void guessUrl() throws Exception{
        assertEquals(MimeTypes.getMimeTypeFromURL("http://www.somewebsite.com/someimage.jpg"), "image/jpeg");
        assertNull(MimeTypes.getMimeTypeFromURL(""));
        assertNull(MimeTypes.getMimeTypeFromURL(null));
    }
}

File: kota/tests/src/com/hendraanggrian/kota/test/webkit/MimeTypesTest.kt
package com.hendraanggrian.kota.test.webkit

import android.annotation.SuppressLint
import android.support.test.runner.AndroidJUnit4
import com.hendraanggrian.kota.content.toMime
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test
import org.junit.runner.RunWith

/**
 * @author Hendra Anggrian (hendraanggrian@gmail.com)
 */
@RunWith(AndroidJUnit4::class)
class MimeTypesTest {

    @Test
    @SuppressLint("SdCardPath")
    @Throws(Exception::class)
    fun guessExtension() {
        assertEquals("/sdcard/home/important.txt".toMime(), "text/plain")
        assertEquals("txt".toMime(), "text/plain")
        assertEquals("jpg".toMime(), "image/jpeg")
        assertNull("".toMime())
    }

    @Test
    @Throws(Exception::class)
    fun guessUrl() {
        assertEquals("http://www.somewebsite.com/someimage.jpg".toMime(), "image/jpeg")
    }
}